plugins/module_utils/job.py

```diff
@@ -122,430 +114,190 @@ def _job_not_found(job_id, owner, job_name, dd_name, ovrr=None):
-    """Generate JSON output string containing Job status info from SDSF.
-    Writes a temporary REXX script to the USS filesystem to gather output.
-
-    Keyword Arguments:
-        job_id {str} -- The job ID to search for (default: {''})
-        owner {str} -- The owner of the job (default: {''})
-        job_name {str} -- The job name search for (default: {''})
-
-    Returns:
-        tuple[int, str, str] -- RC, STDOUT, and STDERR from the REXX script.
-    """
-    get_job_status_rexx = """/* REXX */
-arg options
-parse var options param
-upper param
-parse var param 'JOBID=' jobid ' OWNER=' owner,
-' JOBNAME=' jobname
-
-rc=isfcalls('ON')
-
-jobid = strip(jobid,'L')
-if (jobid <> '') then do
-ISFFILTER='JobID EQ '||jobid
-end
-owner = strip(owner,'L')
-if (owner <> '') then do
-ISFOWNER=owner
-end
-jobname = strip(jobname,'L')
-if (jobname <> '') then do
-ISFPREFIX=jobname
-end
-
-Address SDSF "ISFEXEC ST (ALTERNATE DELAYED)"
-if rc<>0 then do
-Say '[]'
-Exit 0
-end
-if isfrows == 0 then do
-Say '-----NO JOBS FOUND-----'
-end
-else do
-do ix=1 to isfrows
-    if ix<>1 then do
-    end
-    Say '-----START OF JOB-----'
-    Say 'job_id'||':'||value('JOBID'||"."||ix)
-    Say 'job_name'||':'||value('JNAME'||"."||ix)
-    Say 'subsystem'||':'||value('ESYSID'||"."||ix)
-    Say 'system'||':'||value('SYSNAME'||"."||ix)
-    Say 'owner'||':'||value('OWNERID'||"."||ix)
-    Say 'ret_code_msg'||':'||value('RETCODE'||"."||ix)
-    Say 'class'||':'||value('JCLASS'||"."||ix)
-    Say 'content_type'||':'||value('JTYPE'||"."||ix)
-    Say '-----END OF JOB-----'
-end
-end
-
-rc=isfcalls('OFF')
-
-return 0
-"""
-    try:
-        module = AnsibleModuleHelper(argument_spec={})
-        jobid_param = "jobid=" + job_id
-        owner_param = "owner=" + owner
-        jobname_param = "jobname=" + job_name
-
-        tmp = NamedTemporaryFile(delete=True)
-        with open(tmp.name, "w") as f:
-            f.write(get_job_status_rexx)
-        chmod(tmp.name, S_IEXEC | S_IREAD | S_IWRITE)
-        args = [jobid_param, owner_param, jobname_param]
-
-        cmd = [tmp.name, " ".join(args)]
-        rc, out, err = module.run_command(args=cmd)
-    except Exception:
-        raise
-    return rc, out, err
-
-
-def _get_return_code_num(rc_str):
-    """Parse an integer return code from
-    z/OS job output return code string.
+def _parse_steps(job_str):
+    """Parse the dd section of output to retrieve step-wise CC's
 
-    Arguments:
-        rc_str {str} -- The return code message from z/OS job log (eg. "CC 0000")
+    Args:
+        job_str (str): The content for a given dd.
 
     Returns:
-        Union[int, NoneType] -- Returns integer RC if possible, if not returns NoneType
+        list[dict]: A list of step names listed as "step executed" the related CC.
     """
+    stp = []
+    if "STEP WAS EXECUTED" in job_str:
+        pile = re.findall(r"(.*?)\s-\sSTEP\sWAS\sEXECUTED\s-\s(.*?)\n", job_str)
+        for match in pile:
```

Comment:
Use of `pile` did make me smile, I have seen worse, not a deal breaker or anything that needs changing but for readability `for match in steps` or `step in steps` :)

Reply:
bad habit of mine... I've changed pile to steps